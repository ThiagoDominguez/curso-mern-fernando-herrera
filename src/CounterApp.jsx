import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value = 10 }) => {
  const [count, setCount] = useState(value);

  const handleSum = () => {
    setCount(count + 1);
  };
  const handleSubstract = () => {
    setCount(count - 1);
  };

  const handleReset = () => {
    setCount(0);
  };

  return (
    <div>
      <h1>CounterApp</h1>
      <h2>{count}</h2>
      <button onClick={handleSum}>Sumar</button>
      <button onClick={handleSubstract}>Restar</button>
      <button onClick={handleReset}>Resetear</button>
    </div>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
  count: PropTypes.number,
  setCount: PropTypes.func,
};

export default CounterApp;

// * el snippet rafcp crea todo la estructura de este componente , importa react y proptypes, y hace el export default
