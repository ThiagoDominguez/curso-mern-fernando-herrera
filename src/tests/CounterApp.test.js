import React from "react";
import CounterApp from "../CounterApp";
import { shallow } from "enzyme";
{
  import("jest").Config;
}

describe("Pruebas en componente CounterApp", () => {
  // let wrapper;//!Al inicializar esto sin valor pasa a ser undefined , por lo tanto en el scope de cada test perdemos el intelisense de autocompletado
  let wrapper = shallow(<CounterApp />);

  beforeEach(() => {
    //* Esto se ejecuta cada vez que ingresa a una prueba

    wrapper = shallow(<CounterApp />);
    // ! Aqui estariamos inicializando el CounterApp de nuevo por cada prueba
  });

  test("Debe de mostrarse el CounterApp correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar el valor por defecto de 100", () => {
    const wrapper = shallow(<CounterApp value={100} />);

    const valueCount = wrapper.find("h2").text().trim();
    expect(valueCount).toBe("100");
  });

  test("Debe de incrementar al presionar el boton suma", () => {
    wrapper.find("button").at(0).simulate("click");

    const valueCount = wrapper.find("h2").text().trim();

    expect(valueCount).toBe("11");
  });
  test("Debe de decrementar al presionar el boton resta", () => {
    wrapper.find("button").at(1).simulate("click");
    const valueCount = wrapper.find("h2").text().trim();

    expect(valueCount).toBe("9");
  });
  test("Debe de resetear el contador al presionar el boton resetear", () => {
    wrapper.find("button").at(2).simulate("click");
    const valueCount = wrapper.find("h2").text().trim();

    expect(valueCount).toBe("0");
    console.log(valueCount);
  });
});
