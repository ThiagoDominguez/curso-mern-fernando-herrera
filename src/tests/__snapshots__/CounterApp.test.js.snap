// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente CounterApp Debe de mostrarse el CounterApp correctamente 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    Sumar
  </button>
  <button
    onClick={[Function]}
  >
    Restar
  </button>
  <button
    onClick={[Function]}
  >
    Resetear
  </button>
</div>
`;
